tests/unit_tests/test_matchmaker_queue_algorithm.py

```diff
@@ -82,3 +124,36 @@ def test_stable_marriage_unmatch():
 
     assert (s1, s4) in matches  # quality: 0.96622
     assert (s2, s3) in matches  # quality: 0.96623
+
+
+def test_newbies_are_forcefully_matched_with_newbies():
+    newbie1 = Search([p(0, 500, ladder_games=9)])
+    newbie2 = Search([p(2500, 500, ladder_games=9)])
+    pro = Search([p(2500, 10, ladder_games=100)])
+
+    searches = [newbie1, pro, newbie2]
+    matches = algorithm.stable_marriage(searches)
+
+    assert (newbie1, newbie2) in matches or (newbie2, newbie1) in matches
+
+
+def test_unmatched_newbies_forcefully_match_pros():
+    newbie = Search([p(0, 500, ladder_games=0)])
+    pro = Search([p(2500, 10, ladder_games=100)])
+
+    searches = [newbie, pro]
+    matches = algorithm.stable_marriage(searches)
+
+    assert (newbie, pro) in matches or (pro, newbie) in matches
```

Comment:
Hmm. Now I'm not so sure if it's a good idea to match pros with noobs. I think it's common for pros to sit in queue while trying to find an opponent in aeolus. it would be unfortunate for a new player to get matched against one of the top ladder players. Neither person in that match would have a good time.

Maybe instead of doing a completely random match there could be some sort of heuristic like "any player with rating lower than 2 standard deviations plus my mean" or something like that. Maybe it shouldn't even use standard deviation and should just be a hardcoded number say 200?

Either way I think it's probably better to avoid this situation where a complete new player gets matched with a top ladder player. Good test though!

Reply:
Then I'd suggest we only match with with players who have rating mean < 2000? Looking at the "Rating distribution" graph in the 1v1 lobby that should be "basically everyone" :D

Reply:
(side node: I needed a word for "not a newbie in the tests so I called them "pros". I now call everyone with less than ten games a "newbie", everyone with rating > 2500 a "top" player and everyone else a "pro". Maybe they should be `normal_player` instead.)

Reply:
Ok, that’s a good solution too. Note that the ratings in the ladder tab are calculated as `mean - 3 * deviation` so they will be a bit lower than the true skill mean